crates/ruff_python_parser/resources/valid/expressions/dictionary.py

```diff
@@ -31,6 +31,8 @@
 {"a": "b", **c, "d": "e"}
 {1: 2, **{'nested': 'dict'}}
 {x * 1: y ** 2, **call()}
+# Here, `not` isn't allowed but parentheses resets the precedence
+{**(not x)}
```

Comment:
There was a bug with `parse_expression_with_bitwise_or_precedence` where it would report an error even for parenthesized expression. So, this test case verifies that the bug is fixed.